buildSrc/src/main/kotlin/com/datadog/gradle/plugin/jsonschema/PokoDeserializerGenerator.kt

```diff
@@ -176,13 +181,20 @@ class PokoDeserializerGenerator(
             funBuilder.beginControlFlow("if (entry.key !in %L)", RESERVED_PROPERTIES_NAME)
         }
 
-        assignDeserializedProperty(
-            propertyType = additionalProperties,
-            assignee = "${PokoGenerator.ADDITIONAL_PROPERTIES_NAME}[entry.key]",
-            getter = "entry.value",
-            nullable = false,
-            funBuilder = funBuilder
-        )
+        if (additionalProperties is TypeDefinition.Class) {
+            funBuilder.addStatement(
+                "%L[entry.key]=entry.value",
```

Comment:
very nit: missing spaces around `=`, but I don't think it is a big deal in the end :)